src/clj_discord_bot/core.clj

```diff
@@ -1,84 +1,64 @@
-(defn quaggan-answer [data message]
-      (Thread/sleep (* (+ (rand-int 5) 1) 60 1000))
-      (discord/answer-command data message (img-search "quaggan")))
-
-(defn quaggan-joe [type data]
-      (let [message (get data "content")
-            mentions (get data "mentions")
-            mention-all (get data "mention_everyone")]
-           (println "THIS IS A TEST")
-           (println message)
-           (println quaggan-token)
-
-           (if (.contains message quaggan-token)
-             "test")))
-
-(defn gandhi-spellcheck [type data]
-      (let [message (get data "content")]
-           (if (re-find #"(?i)ghandi" message)
-             (discord/answer-command data message "Gandhi  (╯°□°）╯︵ ┻━┻"))))
-
-(defn links-mentioned [type data]
-      (let [message (get data "content")]
-           (if (re-find #"(?i)link" message)
-             (discord/answer-command data message (img-search "lynx")))))
 
 (defn help [type data]
-      (let [message (get data "content")]
-           (print "helpin")
-           (if (.equals "!help" message)
-             (discord/answer-command data message
-                                     (str "Commands" (apply str
-                                          (map #(str "\n" (:doc (meta %))) [#'find-img
-                                                                            #'d20
-                                                                            #'help])))))))
-
+  (discord/answer-command data
+                          (get data "content")
+                          (str "Commands"
+                               (apply str (map #(str "\n" (:doc (meta %)))
+                                               [#'img-search/find-img
+                                                #'roll/d20
+                                                #'summon/game-summon])))))
+
+(defn game_update [type data]
+  (let [server_id 0 ;server id is not returned in message data, so ignore for now ... (get-in data ["guild_id"])
+        user_id (get-in data ["user" "id"])
+        game_name (get-in data ["game" "name"])]
+    (when (and (some? server_id)
+               (some? user_id)
+               (some? game_name))
+      (db/game-insertion server_id
+                         user_id
+                         game_name))))
+
+(defn command-mux [type data]
+  (let [message (get data "content")]
+    (try
+      (cond
+        (.contains message "clojure")
+          (evangelize/get-propaganda type data)
+        (.equals "!help" message)
+          (help type data)
+        (.equals "!d20" message)
+          (roll/d20 type data)
+        (.startsWith message "!summon ")
+          (summon/game-summon type data)
+        (.startsWith message "!img ")
+          (img-search/find-img type data)
+        (re-find #"(?i)ghandi" message)
+          (misc/gandhi-spellcheck type data)
+        (re-find #"(?i)link" message)
+          (misc/links-mentioned type data))
+      (catch Exception e)) ; i still dont care
```

Comment:
we should log out the exceptions somewhere or something so that we know why the bot fucks up